src/main/java/com/yelp/nrtsearch/server/luceneserver/logging/HitsLoggerFetchTask.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.logging;
+
+import com.yelp.nrtsearch.server.grpc.LoggingHits;
+import com.yelp.nrtsearch.server.grpc.SearchResponse;
+import com.yelp.nrtsearch.server.luceneserver.search.FetchTasks.FetchTask;
+import com.yelp.nrtsearch.server.luceneserver.search.SearchContext;
+import com.yelp.nrtsearch.server.plugins.HitsLoggerPlugin;
+import java.util.List;
+
+/**
+ * Implementation of {@link FetchTask} which holds the required context to be able to log hits for a
+ * search request.
+ */
+public class HitsLoggerFetchTask implements FetchTask {
+  private final HitsLogger hitsLogger;
+
+  public HitsLoggerFetchTask(HitsLoggerCreator hitsLoggerCreator, LoggingHits loggingHits) {
```

Comment:
Is there a reason to pass the `HitsLoggerCreator`, it is a static singleton

Reply:
Good catch! I was following what has been done in [HighlightFetchTask here](https://sourcegraph.yelpcorp.com/Yelp/nrtsearch@ef18628bb63e04617cda088cda7531178c8d9236/-/blob/src/main/java/com/yelp/nrtsearch/server/luceneserver/highlights/HighlightFetchTask.java?L49) and missed that I can just call HitsLoggerCreator.getInstance().createHitsLogger(loggingHits) instead